File: dsa_leetcode_beginner/0001-two-sum/0001-two-sum.cpp
#include <unordered_map>
#include <vector>
#include <stdexcept>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];

            if (hashmap.find(complement) != hashmap.end()) {
                return {hashmap[complement], i};
            }

            hashmap[nums[i]] = i;
        }
        throw runtime_error("No two sum solution");
    }
};

// Question: https://leetcode.com/problems/two-sum
// Blog: https://blog.unwiredlearning.com/two-sum

File: dsa_leetcode_beginner/0001-two-sum/0001-two-sum.js
function twoSum(nums, target) {
    const hashmap = new Map();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];

        if (hashmap.has(complement)) {
            return [hashmap.get(complement), i];
        }

        hashmap.set(nums[i], i);
    }
    throw new Error("No two sum solution");
}

// Question: https://leetcode.com/problems/two-sum
// Blog: https://blog.unwiredlearning.com/two-sum